### PR TITLE
add new platform for managing Quanta Switches

### PR DESCRIPTION
**Pull request by Quanta-Switch**
##### SUMMARY
Add new platform qnos for network_cli connection.
##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/network/qnos/qnos.py
lib/ansible/plugins/action/qnos.py
lib/ansible/plugins/cliconf/qnos.py
lib/ansible/plugins/terminal/qnos.py
test/integration/targets/qnos_config

##### ADDITIONAL INFORMATION

```paste below
ansible_network_os=qnos
ansible_connection=network_cli
```

### CODE REVIEW

**Comment by pabelanger** [comment]
This looks to share the same commits on the other 2 PRs, is there a reason for this?

**Comment by chiuhsiapeng** [comment]
Hi Paul,
Thanks for your response.

Yes, this PR shared the same commits on the other 2 PRs (the other PRs depends on this PR)
The reason is I would like to reduce the number of files (< 50).
This part is basic for a new network platform (connection=network_cli)

The other parts are
common parts – for qnos network modules
qnos_command – for qnos_command
The reason is gitbot tells me one module one PR and I have five qnos network modules to upstream.

I am new to Github. I am trying to be familiar with the convention and sorry for the inconveniences.

**Comment by ganeshrn** [comment]
@chiuhsiapeng This PR has a merge commit. Please refer the steps here https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html to resolve the issue.

File: test/sanity/ignore.txt [hunk 1]
```diff
@@ -6914,6 +6914,7 @@ lib/ansible/plugins/action/netconf.py action-plugin-docs # base class for deprec
 lib/ansible/plugins/action/network.py action-plugin-docs # base class for network action plugins
 lib/ansible/plugins/action/normal.py action-plugin-docs # default action plugin for modules without a dedicated action plugin
 lib/ansible/plugins/action/nxos.py action-plugin-docs # base class for deprecated network platform modules using `connection: local`
+lib/ansible/plugins/action/qnos.py action-plugin-docs # base class for deprecated network platform modules using `connection: local`
```
Comment (mattclay) [inline test/sanity/ignore.txt:5711]: Why is new code being added that depends on deprecated functionality?

**Comment by quantasw** [comment]
As the local connection (provider option) is deprecated and is not supported by qnos (followed ganeshrn’s suggestion), no qnos.py exists at plugins/doc_fragments.
Adding this line to skip the checking ‘ansible-test sanity --test action-plugin-docs.’

File: docs/docsite/rst/network/user_guide/platform_qnos.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+   ansible_user: myuser
+   ansible_password: !vault...
+
+
+- If you are using SSH keys (including an ssh-agent) you can remove the ``ansible_password`` configuration.
+- If you are accessing your host through a bastion/jump host, you cannot include your SSH password in the ``ProxyCommand`` directive. To prevent secrets from leaking out (for example in ``ps`` output), SSH does not support providing passwords via environment variables.
+
+Options
+-------
+- See L(cli_command module,../../modules/cli_command_module.html).
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_qnos.rst]:
```suggestion
- See :ref:`cli_command <cli_command_module>.
```

File: docs/docsite/rst/network/user_guide/platform_qnos.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+- See L(cli_config module,../../modules/cli_config_module.html).
```
Comment (samccann) [inline docs/docsite/rst/network/user_guide/platform_qnos.rst]:
```suggestion
-- See :ref:`cli_config <cli_config_module>.
```

File: test/sanity/ignore.txt (hunk 1 above)
Comment (Quanta-Switch) [reply test/sanity/ignore.txt:5711]: As the local connection (provider option) is deprecated and is not supported by qnos (followed ganeshrn’s suggestion), no qnos.py exists at plugins/doc_fragments.
Adding this line to skip the checking ‘ansible-test sanity --test action-plugin-docs.’

**Pull request closed without merging**